src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -494,12 +463,21 @@ public void testWithCacheWithNoViolation() throws Exception {
             new HashSet<>(), Thread.currentThread().getContextClassLoader());
         checker.setModuleFactory(factory);
         checker.configure(createCheckConfig(TranslationCheck.class));
-        checker.setCacheFile(temporaryFolder.newFile().getPath());
+
+        final File cacheFile = temporaryFolder.newFile();
+        checker.setCacheFile(cacheFile.getPath());
+
         checker.setupChild(createCheckConfig(TranslationCheck.class));
         final File file = temporaryFolder.newFile("file.java");
         final List<File> files = new ArrayList<>();
         files.add(file);
         checker.process(files);
+
+        final Properties cache = new Properties();
+        cache.load(Files.newBufferedReader(cacheFile.toPath()));
+
+        // There should be no objects in cache since there are no violations and config is not set.
+        assertTrue(cache.isEmpty());
```

Comment:
I'm not sure if I didn't notice this before, but this test is a lie.
`PropertyCacheFile` can't be instantiated without a configuration, it will throw an exception.
The whole reason the cache is empty is because `destroy` is never called to commit the changes because we play with `Checker` directly instead of going through a 'normal' lifecycle.

We specifically set the configuration with the line `checker.configure(createCheckConfig(TranslationCheck.class));`. So it is a lie there is no configuration.
Also if there are no violations, cache is still filled with all files checked.

Is this test relying on some code coverage? If so, let's rework this test around that.